docs/framework/network-programming/ntlm-and-kerberos-authentication.md

```diff
@@ -26,31 +26,53 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
 Default NTLM authentication and Kerberos authentication use the Microsoft Windows user credentials associated with the calling application to attempt authentication with the server. When using non-default NTLM authentication, the application sets the authentication type to NTLM and uses a <xref:System.Net.NetworkCredential> object to pass the user name, password, and domain to the host, as shown in the following example.  
   
 ```vb  
-Dim MyURI As String = "http://www.contoso.com/"  
-Dim WReq As WebRequest = WebRequest.Create(MyURI)  
-WReq.Credentials = _  
-    New NetworkCredential(UserName, SecurelyStoredPassword, Domain)  
+Dim myUri As String = "http://www.contoso.com/"  
+Using handler As New HttpClientHandler()
+    With handler
+        .Credentials = New NetworkCredential(UserName, SecurelyStoredPassword, Domain)
+    End With
+    Using client As New HttpClient(handler)
+        Dim result As String = client.GetStringAsync(myUri).GetAwaiter().GetResult()
```

Comment:
I don't know anything about VB, but based on [these docs](https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/concepts/async/#BKMK_HowtoWriteanAsyncMethod) it seems like we should be able to use Await here as well (so we don't accidentally promote "sync over async" patterns).

Reply:
I agree if that is available. I didn't know for sure.